pkg/kubectl/cmd/delete.go

```diff
@@ -248,13 +261,20 @@ func ReapResult(r *resource.Result, f cmdutil.Factory, out io.Writer, isDefaultD
 		if err != nil {
 			return err
 		}
+		// Set WaitForDeletion to false for namespace deletion unless it was explicitly set by the user.
+		if info.Mapping.Resource == "Namespace" && isDefaultWaitForDeletion {
```

Comment:
> Kubectl does not wait for namespace deletion, while it waits for all others (example: deployment, replicasets).

I don't think that's accurate... kubectl didn't wait for deletion before for server-side finalized resources... pods and namespaces just happened to be the only ones that weren't deleted synchronously.